src/modules/utils/motordrivercontrol/drivers/TMC220X/TMC220X.cpp

```diff
@@ -0,0 +1,1870 @@
+    this->sgthrs_register_value = 0;
+    this->coolconf_register_value = TMC220X_ZEROS_DEFAULT_DATA;
+    this->chopconf_register_value = TMC220X_CHOPCONF_DEFAULT_DATA;
+    this->pwmconf_register_value = TMC220X_PWMCONF_DEFAULT_DATA;
+    
+
+    // Configure SpreadCycle (also uses in StealthChop when TPWMTHRS is set)
+    // (default values come from the Quick Configuration Guide chapter in TMC2208 datasheet)
+    int8_t toff  = THEKERNEL->config->value(motor_driver_control_checksum, cs, spreadcycle_toff_checksum)->by_default(5)->as_int();
+    int8_t tbl   = THEKERNEL->config->value(motor_driver_control_checksum, cs, spreadcycle_tbl_checksum)->by_default(32)->as_int();
+    int8_t hstrt = THEKERNEL->config->value(motor_driver_control_checksum, cs, spreadcycle_hstrt_checksum)->by_default(4)->as_int(); 
+    int8_t hend  = THEKERNEL->config->value(motor_driver_control_checksum, cs, spreadcycle_hend_checksum)->by_default(0)->as_int(); 
+
+    setSpreadCycleChopper(toff, tbl, hstrt, hend);
+
+    // Configure StealthChop
+    uint8_t lim       = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_lim_checksum)->by_default(12)->as_int();
+    uint8_t reg       = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_reg_checksum)->by_default(1)->as_int();
+    uint8_t freewheel = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_freewheel_checksum)->by_default(0)->as_int();
+    bool autograd     = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_autograd_checksum)->by_default(true)->as_bool();
+    bool autoscale    = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_autoscale_checksum)->by_default(true)->as_bool(); 
+    uint8_t freq      = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_freq_checksum)->by_default(1)->as_int();
+    uint8_t grad      = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_grad_checksum)->by_default(0)->as_int();
+    uint8_t ofs       = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_pwm_ofs_checksum)->by_default(36)->as_int();
+    int tpwmthrs      = THEKERNEL->config->value(motor_driver_control_checksum, cs, stealthchop_tpwmthrs_checksum)->by_default(0)->as_int();
+
+    setStealthChop(lim, reg, freewheel, autograd, autoscale, freq, grad, ofs);
+
+    if (tpwmthrs > 0) {
+        setStealthChopthreshold(tpwmthrs);
+    }
+
+    // Select mode (0 - spreadCycle, 1 - stealthChop)
+    if (this->chopper_mode == 1) {
+        setSpreadCycleEnabled(false);
+    } else {
+        setSpreadCycleEnabled(true);
+    }
+    
+    // TMC2209 specific config
+    if (chip_type == StepstickParameters::CHIP_TYPE::TMC2209) {
+        
+        uint8_t tcoolthrs   = THEKERNEL->config->value(motor_driver_control_checksum, cs, stallguard_tcoolthrs_checksum)->by_default(0)->as_int();
```

Comment:
This value is a possible 20 bits, and will overflow a `uint8_t` - suggest using `int`